src/hotspot/share/gc/g1/g1GCPhaseTimes.cpp

```diff
@@ -331,7 +331,7 @@ void G1GCPhaseTimes::log_phase(WorkerDataArray<double>* phase, uint indent_level
 
   for (uint i = 0; i < phase->MaxThreadWorkItems; i++) {
     WorkerDataArray<size_t>* work_items = phase->thread_work_items(i);
-    if (work_items != NULL) {
+    if (work_items != NULL && indent(indent_level + 1) != NULL) {
```

Comment:
No, don't do this.  `indent` never returns NULL.  It asserts the given argument is in bounds.  The warning is a (gcc9-specific?) false positive, similar to JDK-8235819.

Reply:
> No, don't do this. `indent` never returns NULL. It asserts the given argument is in bounds. The warning is a (gcc9-specific?) false positive, similar to JDK-8235819.

Thanks @kimbarrett for your review.

Yes, I agree that this seems to be false positive with gcc 9.

Fastdebug build wouldn't fail due to the existence of the assert.
But release vm would fail since there is no such an assert.

So any suggestions?
Thanks.

Reply:
The string version ("%*s") of mine is better as it will correctly indent the zero indentation case.

Reply:
> Maybe instead use something like:
> `out->print("%*s%s", 2 * indent_level, "", "indented*s");`
> or:
> `out->print("%*c%s", 2 * indent_level, ' ', "indented*c");`
> 
> And get rid of the "Indents" string array and associated functions.

Good suggestion!

Thanks @lkorinth .
Will update it later if there is no objection.

Reply:
https://github.com/openjdk/jdk/pull/3934 solves the same problem via out->sp(indent_level * 2); please coordinate the fix with @tstuefe, and whatever solution you chose to use, please remove all of the old usages.

Reply:
> #3934 solves the same problem via out->sp(indent_level * 2); please coordinate the fix with @tstuefe, and whatever solution you chose to use, please remove all of the old usages of indent().

Okay.
I didn't know @tstuefe 's PR before.

I'll close mine if @tstuefe would like to fix this issue.
Thanks.

Reply:
Or do it the other way, yours is correctly linked to 8266773. Sync with Thomas so that we only have one pull request.

Reply:
> Or do it the other way, yours is correctly linked to 8266773. Sync with Thomas so that we only have one pull request.

I prefer @tstuefe 's `out->sp(indent_level * 2)`.

Patch has been updated.

Testing:
  - test/hotspot/jtreg/gc on Linux/x64
 
Thanks.